### PR TITLE
Raise exception if torch.func.* calls torch.compile functions

### LINKED ISSUE #128711: vmap fails to call torch.compiled function
### Reproducer

```python
import torch

@torch.compile(backend="aot_eager", fullgraph=True)
def fn(x):
    return x.sin()

x = torch.randn(3, 4)
y = torch.func.vmap(fn)(x)

# ...
# torch/_dynamo/convert_frame.py:178: in _fn
#     return fn(*args, **kwargs)
# torch/_dynamo/convert_frame.py:564: in transform
#     tracer = InstructionTranslator(
# torch/_dynamo/symbolic_convert.py:2396: in __init__
#     self._throw_if_in_functorch()
# torch/_dynamo/symbolic_convert.py:2452: in _throw_if_in_functorch
#     unimplemented(msg)
# torch/_dynamo/exc.py:221: in unimplemented
#     raise Unsupported(msg)
# E   torch._dynamo.exc.Unsupported: torch.func.vmap(fn) requires the function to be inlined by dynamo
# E
# E   Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information
# E
# E
# E   You can suppress this exception and fall back to eager by setting:
# E       import torch._dynamo
# E       torch._dynamo.config.suppress_errors = True
```

We incorrectly exit when we detect that we're inside a functorch in InstructionTranslator. The 

### Versions

main

cc @zou3519 @ezyang @chauhang @penguinwu @bdhirsh @anijain2305

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128736

cc @zou3519 @Chillee @samdow @kshitij12345 @janeyx99 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -2445,10 +2445,22 @@ def _throw_if_in_functorch(self):
             torch._C._functorch.TransformType.Grad,
             torch._C._functorch.TransformType.Jvp,
         )
+
+        # Is there a better way to detect this?
+        from_eager = counters.get("graph_break", None) is None
```
Comment (guilhermeleobas) [inline torch/_dynamo/symbolic_convert.py:2450]: This is 100% wrong.

File: test/dynamo/test_higher_order_ops.py [hunk 2]
```diff
@@ -5285,6 +5285,45 @@ def fn(x):
...
+
+        x = torch.randn(3, 4)
+        fn = torch.compile(backend="aot_eager", fullgraph=True)(wrapped_fn)
+
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.Unsupported,
+            "Calling torch.func.vmap\\(compiled_fn\\) function from eager mode is not supported",
+        ):
+            torch.func.vmap(fn)(x)
+
```
Comment (zou3519) [inline test/dynamo/test_higher_order_ops.py:5300]: This graph breaks today:
```
import torch

def wrapped_fn(x):
    return x.sin()

x = torch.randn(3, 4)
fn = torch.compile(backend="aot_eager")(wrapped_fn)

torch.vmap(fn)(x)
```

File: torch/_dynamo/symbolic_convert.py (hunk 1 above)
Comment (guilhermeleobas) [reply torch/_dynamo/symbolic_convert.py:2450]: If one of the inputs is a functorch tensor and the backend is not `eager`, raise a graph break.

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**